sylvia-derive/src/message.rs

```diff
@@ -851,7 +852,7 @@ impl<'a> GlueMessage<'a> {
         let segments = &module.segments;
         assert!(!segments.is_empty());
 
-        let syn::PathSegment { ident, .. } = &segments[0];
+        let syn::PathSegment { ident, .. } = &segments.last().unwrap();
```

Comment:
I changed this method to accept `ContractMessageAttr`. `segments` should never be empty as in such case `parse` method will fail. Error in such case would be `expected identifier` which is not very informative and it might be better to parse this error to point into proper attribute structure.